q2_vizard/_util.py

```diff
@@ -60,3 +63,39 @@ def _measure_validation(metadata, measure):
                 f' be parsed by this visualization. Please remove `{char}`'
                 ' from this Metadata column name.'
             )
+
+
+# manual checks for the rendered stats in boxplot
+def _calculate_median(sorted_values):
+    n = len(sorted_values)
+    middle = n // 2
+
+    if n % 2 == 0:
+        median = (sorted_values.iloc[middle - 1] +
+                  sorted_values.iloc[middle]) / 2.0
+    else:
+        median = sorted_values.iloc[middle]
+    return median
+
+
+def _calculate_quartiles(sorted_values):
+    q1 = np.percentile(sorted_values, 25, interpolation='linear')
+    q3 = np.percentile(sorted_values, 75, interpolation='linear')
+    return q1, q3
+
+
+def _calculate_percentile(sorted_values, percentile):
```

Comment:
Could the tests also use `np.percentile` instead of this function?

It seems like maybe a bit more is happening here, but I'm not sure.

Reply:
yep, you're totally right @ebolyen - i had originally tried to go without an external library and things got a bit more involved (hence me templating these calculations out into helper methods), but then i ended up using numpy after all. i've put all of these calculations inline and am now just using numpy's methods.